embassy-rp/src/pwm.rs

```diff
@@ -0,0 +1,363 @@
+
+        unsafe {
+            let p = inner.regs();
+            if config.divider > FixedU16::<fixed::types::extra::U4>::from_bits(0xFF_F) {
+                panic!("Requested divider is too large");
+            }
+
+            p.csr().write(|w| {
+                w.set_divmode(config.divmode);
+                w.set_a_inv(config.invert_a);
+                w.set_b_inv(config.invert_b);
+                w.set_ph_correct(config.phase_correct);
+                w.set_en(config.enable);
+            });
+            p.div().write_value(ChDiv(config.divider.to_bits() as u32));
+            p.cc().write(|w| {
+                w.set_a(config.compare_a);
+                w.set_b(config.compare_b);
+            });
+            p.top().write_value(ChTop(config.top as u32));
+
+            if let Some(pin) = &a {
+                pin.io().ctrl().write(|w| w.set_funcsel(4));
+            }
+            if let Some(pin) = &b {
+                pin.io().ctrl().write(|w| w.set_funcsel(4));
+            }
+        }
+        Self { inner, _mode: mode }
+    }
+
+    pub fn phase_advance(&mut self) {
+        let p = self.inner.regs();
+        unsafe {
+            p.csr().modify(|w| w.set_ph_adv(true));
+            let wait_more = |r: ChCsr| r.en() && r.ph_adv();
+            while wait_more(p.csr().read()) {}
+        }
+    }
+
+    pub fn phase_retard(&mut self) {
+        let p = self.inner.regs();
+        unsafe {
+            p.csr().modify(|w| w.set_ph_ret(true));
+            let wait_more = |r: ChCsr| r.en() && r.ph_ret();
+            while wait_more(p.csr().read()) {}
+        }
+    }
+
+    pub fn counter(&self) -> u16 {
+        unsafe { self.inner.regs().ctr().read().ctr() }
+    }
+
+    fn bit(&self) -> u32 {
+        1 << self.inner.number() as usize
+    }
+}
+
+impl<'d, T: Channel> Pwm<'d, T, Blocking> {
+    pub fn new_blocking_a(
+        inner: impl Peripheral<P = T> + 'd,
+        a: impl Peripheral<P = impl PwmPin<T> + 'd> + 'd,
+        config: Config,
+    ) -> Self {
+        into_ref!(a);
+        Self::new_inner(inner, Some(a.map_into()), None, config, Blocking)
+    }
+
+    pub fn new_blocking_b(
+        inner: impl Peripheral<P = T> + 'd,
+        b: impl Peripheral<P = impl PwmPin<T> + 'd> + 'd,
+        config: Config,
+    ) -> Self {
+        into_ref!(b);
+        Self::new_inner(inner, None, Some(b.map_into()), config, Blocking)
+    }
+
+    pub fn new_blocking_ab(
+        inner: impl Peripheral<P = T> + 'd,
+        a: impl Peripheral<P = impl PwmPin<T> + 'd> + 'd,
+        b: impl Peripheral<P = impl PwmPin<T> + 'd> + 'd,
+        config: Config,
+    ) -> Self {
+        into_ref!(a, b);
+        Self::new_inner(inner, Some(a.map_into()), Some(b.map_into()), config, Blocking)
+    }
+
+    pub fn into_async(self) -> Pwm<'d, T, Async> {
+        Pwm {
+            _mode: Async::enable(self.inner.number()),
+            inner: self.inner,
+        }
+    }
+
+    pub fn wrap<'f>(&'f mut self) {
```

Comment:
perhaps a more descriptive name like `wait_for_wrap()` would be better.

Also, the lifetime is not needed.